checker_arithmetic_test.go

```diff
@@ -0,0 +1,159 @@
+// Licensed under the MIT license, see LICENSE file for details.
+
+package quicktest_test
+
+import (
+	"fmt"
+	"math/big"
+	"testing"
+
+	"github.com/frankban/quicktest"
+)
+
+// TestBinaryArithmeticCheckerMatrix verifies that the binary arithmetic checker
+// correctly handles all possible types of input arguments (got/reference).
+func TestBinaryArithmeticCheckerMatrix(t *testing.T) {
+	numerics := []interface{}{
+		int(99),
+		int8(99),
+		int16(99),
+		int32(99),
+		int64(99),
+		uint(99),
+		uint8(99),
+		uint16(99),
+		uint32(99),
+		uint64(99),
+		float32(99),
+		float64(99),
+	}
+
+	asBigFloat := big.NewFloat(99)
+
+	for _, got := range numerics {
+		for _, reference := range numerics {
+			t.Run(fmt.Sprintf("%T and %T", got, reference), func(t *testing.T) {
+				checker := quicktest.NewBinaryArithmeticChecker(func(value, reference *big.Float) error {
+					if value.Cmp(asBigFloat) != 0 {
+						t.Fatal("unexpected value")
+					}
+					if reference.Cmp(asBigFloat) != 0 {
+						t.Fatal("unexpected value")
+					}
+					return nil
+				})
+
+				err := checker.Check(got, []interface{}{reference}, func(key string, value interface{}) {})
+				if err != nil {
+					t.Fatal("expected nil error from checker")
+				}
+			})
+		}
+	}
+}
+
+func TestBinaryArithmeticCheckerValidation(t *testing.T) {
```

Comment:
Do we need all these tests? Isn't one test for the value and one for the reference sufficient? Also, can this be just part of the checker tests?